src/microbe_stage/CompoundBag.cs

```diff
@@ -10,29 +10,36 @@
 [UseThriveSerializer]
 public class CompoundBag : IEnumerable<KeyValuePair<Compound, float>>
 {
-    /// <summary>
-    ///   The max amount of any compound that can be stored
-    /// </summary>
-    public float Capacity;
-
     [JsonProperty]
     private readonly HashSet<Compound> usefulCompounds = new HashSet<Compound>();
 
+    private float capacity;
+
     /// <summary>
     ///   Creates a new bag
     /// </summary>
     /// <param name="capacity">Specifies the initial capacity of the compound bag</param>
     public CompoundBag(float capacity)
     {
-        Capacity = capacity;
+        this.capacity = capacity;
+    }
+
+    /// <summary>
+    ///   The max amount of any compound that can be stored
+    /// </summary>
+    [JsonProperty]
+    public virtual float Capacity
+    {
+        get => capacity;
+        set => capacity = value;
```

Comment:
I kinda don't like this, I'd prefer if you instead made an ICompoundBag interface that has the common stuff needed and this implements that interface then.